packages/web/src/itinerary/plan.ts

```diff
@@ -0,0 +1,81 @@
+import Event from "./event";
+
+export default class Plan {
+    private _name: string;
+    private _events: Event[] = [];
+
+    constructor(name: string) {
+        this._name = name;
+    }
+
+    /**
+     * Gets the name of the plan
+     */
+    get name(): string {
+        return this._name;
+    }
+
+    /**
+     * Sets the name of the plan
+     */
+    set name(name: string) {
+        this._name = name;
+    }
+
+    /**
+     * Gets number of events in plan
+     */
+    get length(): number {
+        return this._events.length;
+    }
+
+    /**
+     * Gets all the events in a plan
+     */
+    get allEvents(): Event[] {
+        return this._events;
+    }
+
+    /**
+     * Adds a single event, returns true if successful
+     */
+    public addEvent(event: Event): boolean {
+        if (this._events.includes(event)) {
+            return false;
+        }
+        this._events.push(event);
+        return true;
+    }
+
+    /**
+     * Gets a single event
+     */
+    public getEvent(id: number): Event {
+        for (let counter = 0; counter < this._events.length; counter++) {
+            const result = this._events.filter((item) => item.id == id)[0];
```

Comment:
`.filter()` can return an empty array if the item isn't found, which means getting the first element via `[0]` will result in `undefined` (index out of bounds).  We're also looking for a single event so [`.find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) would be a better option here.  Still need to make sure we're not returning undefined.

Reply:
How do you now throw an undefined?